### ISSUE TITLE
Improve TWAP Queries to not require pool id

### ISSUE THREAD

**Issue by mattverse** [osmosis-labs/osmosis#5438]
## Background

Current TWAP Queries, regardless of the quote denom and base denom, require specification of pool id in the query(https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/twap/v1beta1/query.proto#L37-L58)
This is unnecessary for the base asset - quote asset pairs that have an obvious or a single twap pair.

## Suggested Design
Suggested design is that we make the pool id optional for the query, when a query has been given, check in state if we have one pair for the given base asset and quote asset pair. If so, great we use that pool as the twap and return results, if not and multiple pools registered for the base asset and quote asset pair exists, return an error specifying that pool id should be specified. 

## Acceptance Criteria

- Improve TWAP Queries to not require pool id
- Test cases have been added

**Comment by mattverse** [comment]
@AlpinYukseloglu yeah, right now, anyone trying to query twap needs to query or know before hand on the pool id as well as the base and quote denom, which creates an extra overhead on the queries

**Comment by AlpinYukseloglu** [comment]
I see, thanks!
